ballerina/tests/listener_oauth2_provider_test.bal

```diff
@@ -545,3 +545,24 @@ isolated function testTokenIntrospectionRequestWithSecureSocketAndWithHttpUrlSch
     test:assertEquals(response?.jti, "JlbmMiOiJBMTI4Q0JDLUhTMjU2In");
     test:assertEquals(response?.jti, "JlbmMiOiJBMTI4Q0JDLUhTMjU2In");
 }
+
+// Test that the `exp` field is correctly parsed as an integer when given an example string value for `exp` field
+@test:Config {}
+isolated function testStringExpParsing() {
+    json simulatedResponse = {
+        "active": true,
+        "exp": "1672531199",
+        "scope": "read write",
+        "client_id": "test_client_id",
+        "username": "test_user"
+    };
+
+    IntrospectionResponse response = prepareIntrospectionResponse(simulatedResponse);
+
+
+    test:assertTrue(response.active);
+    test:assertEquals(response.exp, 1672531199);
+    test:assertEquals(response.scope, "read write");
+    test:assertEquals(response.clientId, "test_client_id");
+    test:assertEquals(response.username, "test_user");
+}
```

Comment:
```suggestion
@test:Config {}
isolated function testPrepareIntrospectionResponseWithStringExpClaim() {
    json simulatedResponse = {
        active: true,
        exp: "1672531199",
        scope: "read write",
        client_id: "test_client_id",
        username: "test_user"
    };
    IntrospectionResponse response = prepareIntrospectionResponse(simulatedResponse);
    test:assertEquals(response.exp, 1672531199);
}
```
No need to test other fields, just validating the exp claim is enough here, and let's change the function name to suggested one or more descriptive one.